### PR TITLE
Fix gtest fetch issue

### PR DESCRIPTION
**Pull request by fernride-jissing**
bazel fetch errors can also cause issues with bazel queries and bazel fetch can detect dependency changes if run in CI.

### CODE REVIEW

File: repositories/ros2_repositories_impl.bzl [hunk 1]
```diff
@@ -62,7 +62,7 @@ def ros2_repositories_impl():
     maybe(
         http_archive,
         name = "iceoryx",
-        strip_prefix = "iceoryx-2.0.2",
+        strip_prefix = "iceoryx-2.0.3",
```
Comment (mvukov) [inline repositories/ros2_repositories_impl.bzl:65]: This is seems to be unrelated to this PR.

**Review by mvukov — COMMENTED** [review]
TBH, I don't understand what doesn't work for you that requires this gtest patch. Can you elaborate, please?

File: repositories/ros2_repositories_impl.bzl (hunk 1 above)
Comment (mvukov) [reply repositories/ros2_repositories_impl.bzl:65]: Fixed in https://github.com/mvukov/rules_ros2/pull/118.

**Comment by fernride-jissing** [comment]
> TBH, I don't understand what doesn't work for you that requires this gtest patch. Can you elaborate, please?

Sorry for the late reponse. It's an issue that's simple to reproduce:

```bash
git clone git@github.com:mvukov/rules_ros2.git
cd rules_ros2
bazel fetch //...
```

It will also cause issues if one want so run `bazel query` that includes external dependencies.

**Comment by mvukov** [comment]
OK, thanks for the explanation. I'd propose https://github.com/mvukov/rules_ros2/pull/120 as an alternative. I consider patches as a last resort, typically. Please let me know what you think.

**Comment by fernride-jissing** [comment]
> OK, thanks for the explanation. I'd propose #120 as an alternative. I consider patches as a last resort, typically. Please let me know what you think.

Looks good to me! Thanks :)
